govcd/extension.go

```diff
@@ -5,50 +5,63 @@
 package govcd
 
 import (
+	"bytes"
+	"encoding/xml"
 	"errors"
 	"fmt"
-	"net/url"
 
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
 	"github.com/vmware/go-vcloud-director/v2/util"
 )
 
 func GetExternalNetworkByName(vcdClient *VCDClient, networkName string) (*types.ExternalNetworkReference, error) {
```

Comment:
I would like that we follow our current API behaviour and return external network struct vs `ExternalNetworkReference`.

Reply:
Ok just to make sure we're on the same page - you're asking me for a breaking change here, right?  Specifically, someone using this library would bump the version of this lib in their app and the code may not compile due to a type mismatch.

Is there a versioning strategy for this repo?  I couldn't find anything in the markdown files.  As a generalization though, I'll refer to the golang FAQ:

> Packages intended for public use should try to maintain backwards compatibility as they evolve.

and

> If different functionality is required, add a new name instead of changing an old one.

https://golang.org/doc/faq#get_version

Reply:
Well if this function existed - we have to support it due backwards compatibility. Solution is easy, deprecate this function and let's create wrapping function which will return network struct.

Reply:
P.s. Regarding versioning - we follow https://semver.org/

Reply:
Ok, so maybe I'm not quite following what you mean, but that's pretty much how the code looks now.
```go
	externalNetwork := NewExternalNetwork(&vcdClient.Client)
	err := externalNetwork.GetByName(networkName)
```
Would you prefer I put those two lines in new method?

Reply:
Ok, so maybe I'm not quite following what you mean, but that's pretty much how the code looks now.
```go
	externalNetwork := NewExternalNetwork(&vcdClient.Client)
	err := externalNetwork.GetByName(networkName)
```
Would you prefer I put those two lines in new method?

Reply:
In terraform you used a little different approach
````
externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
externalNetwork.ExternalNetwork.HREF = externalNetworkRef.HREF
err = externalNetwork.Refresh()
```
Aim is to avoid such constructions, cause user need good knowledge of govcd for that. So answer is yes - wrapper could just has a few lines which allows to return right construct with externalNetwork which can be used directly with API functions.

Reply:
Sure sounds good.  I'm not too particular on the name, so let me know if you'd like to change it.

Reply:
Well, this one is hard :) - Let's do it simple -> "getExternalNetwork"

Reply:
Sweet can do.  I think it needs to be a public method, no?

Reply:
Yes, definitely :) ant let's add `// DEPRECATED please use getExternalNetwork function instead` for old one